src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -311,13 +311,32 @@ public StringIndirectionNode StringIndirection(string data)
             return _stringIndirectionNodes.GetOrAdd(data);
         }
 
-        public ArrayOfEmbeddedDataNode GCStaticsRegion = new ArrayOfEmbeddedDataNode("__GCStaticRegionStart", "__GCStaticRegionEnd", null);
-        public ArrayOfEmbeddedDataNode ThreadStaticsRegion = new ArrayOfEmbeddedDataNode("__ThreadStaticRegionStart", "__ThreadStaticRegionEnd", null);
-        public ArrayOfEmbeddedDataNode StringTable = new ArrayOfEmbeddedDataNode("__str_fixup", "__str_fixup_end", null);
+        public string GetNodeAliasName(Object node) // TODO: What should the type argument be?
```

Comment:
This is still not much more systematic than the previous solution. In particular, a node can only have one alias.
cc @davidwrighton